admin/client/src/components/form-builder/index.js

```diff
@@ -119,6 +151,101 @@ export class FormBuilderComponent extends SilverStripeComponent {
   }
 
   /**
+   * Update handler passed down to each form field as a prop.
+   * Form fields call this method when their state changes.
+   *
+   * You can pass an optional callback as the third param. This can be used to
+   * implement custom behaviour. For example you can use `createFn` hook from
+   * your controller context like this.
+   *
+   * controller.js
+   * ...
+   * detailEditFormCreateFn(Component, props) {
+   *   const extendedProps = Object.assign({}, props, {
+   *     handleFieldUpdate: (event, updates) => {
+   *       props.handleFieldUpdate(event, updates, (formId, updateFieldAction) => {
+   *         const customUpdates = Object.assign({}, updates, {
+   *           value: someCustomParsing(updates.value),
+   *         };
+   *
+   *         updateFieldAction(formId, customUpdates);
+   *       });
+   *     },
+   *   });
+   *
+   *   return <Component {...extendedProps} />;
+   * }
+   * ...
+   *
+   * @param {object} event - Change event from the form field component.
+   * @param {object} updates - Values to set in state.
+   * @param {string} updates.id - Field ID. Required to identify the field in the store.
+   * @param {function} [fn] - Optional function for custom behaviour. See example in description.
```

Comment:
Square brackets are for optional params http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values